modules/planning/planning_component.cc

```diff
@@ -56,44 +71,89 @@ bool PlanningComponent::Init() {
 
   planning_base_->Init(config_);
 
-  routing_reader_ = node_->CreateReader<RoutingResponse>(
-      FLAGS_routing_response_topic,
-      [this](const std::shared_ptr<RoutingResponse>& routing) {
-        AINFO << "Received routing data: run routing callback."
-              << routing->header().DebugString();
-        std::lock_guard<std::mutex> lock(mutex_);
-        routing_.CopyFrom(*routing);
-      });
-  traffic_light_reader_ = node_->CreateReader<TrafficLightDetection>(
-      FLAGS_traffic_light_detection_topic,
-      [this](const std::shared_ptr<TrafficLightDetection>& traffic_light) {
-        ADEBUG << "Received traffic light data: run traffic light callback.";
-        std::lock_guard<std::mutex> lock(mutex_);
-        traffic_light_.CopyFrom(*traffic_light);
-      });
-
-  pad_msg_reader_ = node_->CreateReader<PadMessage>(
-      FLAGS_planning_pad_topic,
-      [this](const std::shared_ptr<PadMessage>& pad_msg) {
-        ADEBUG << "Received pad data: run pad callback.";
-        std::lock_guard<std::mutex> lock(mutex_);
-        pad_msg_.CopyFrom(*pad_msg);
-      });
+  if (config_.has_adc_id()) {
+    std::string channel_id = "/" + std::to_string(config_.adc_id());
 
-  if (FLAGS_use_navigation_mode) {
-    relative_map_reader_ = node_->CreateReader<MapMsg>(
-        FLAGS_relative_map_topic,
-        [this](const std::shared_ptr<MapMsg>& map_message) {
-          ADEBUG << "Received relative map data: run relative map callback.";
+    routing_reader_ = node_->CreateReader<RoutingResponse>(
+        FLAGS_routing_response_topic + channel_id,
```

Comment:
channel_id can be default to "", so you can always write FLAGS_routing_response_topic + channel_id, and don't need to have similar reader/writer assignment for with and without adc_id

Reply:
good point